### PR TITLE
Unfold before termination check!

### PR DESCRIPTION
**Pull request by wsx-ucb**
Really not sure what is the best solution here.
The current implementation is very hacky...
Based on #915.

### CODE REVIEW

**Review by ice1000 — COMMENTED** [review]
Could you please also update the stdlib with `pmap`?

**Comment by wsx-ucb** [comment]
Updated the `pmap` usage.
I also attempted to switch to using `pmapd` in `Data/Vec.aya`, but it seems to lead to an internal exception that is unrelated to `terck`.
I marked this later usage as a todo.

File: base/src/main/java/org/aya/terck/CallResolver.java [hunk 1]
```diff
@@ -151,7 +153,7 @@ private Relation compareConArgs(@NotNull ImmutableSeq<Arg<Term>> conArgs, @NotNu
 
   @Override public @NotNull Term pre(@NotNull Term term) {
     // TODO: Rework error reporting to include the original call
-    // term = whnf(term);
+    term = new Expander.ConservativeWHNFer(new TyckState(factory), ImmutableSet.of(caller.ref)).apply(term);
```
Comment (ice1000) [inline base/src/main/java/org/aya/terck/CallResolver.java:156]: Should it be `targets` instead of `caller.ref`?

Comment (wsx-ucb) [reply base/src/main/java/org/aya/terck/CallResolver.java:156]: Oh good point, in principle we should avoid unfolding any recursive function in the current SCC, and `target` contains all of them, right?

Comment (ice1000) [reply base/src/main/java/org/aya/terck/CallResolver.java:156]: I think so. Cc @imkiva

**Comment by ice1000** [comment]
Could you please also replace set truncation's eliminator with pmap?

**Review by ice1000 — APPROVED** [review]
bors d=wsx-ucb

**Review by imkiva — APPROVED** [review]
bors d=wsx-ucb

**Comment by ice1000** [comment]
bors merge

**Pull request merged**